packages/option-t/src/plain_result/try_catch.ts

```diff
@@ -41,3 +43,16 @@ export function tryCatchIntoResultWithEnsureError<T>(producer: ProducerFn<T>): R
         return errWrapped;
     }
 }
+
+export function tryCatchIntoResultAndMapErrFallback<T>(producer: ProducerFn<T>): Result<T, Error> {
```

Comment:
We need to write:

1. Why we add this.
2. Why this don't wrap 
3. We may deprecate this in the future and begin to recommend to switch `tryCatchIntoResultWithEnsureError()`